updateLib.js

```diff
@@ -0,0 +1,177 @@
+
+  return {
+    file,
+    variant,
+    fileNodeVersion,
+    nodeMajorVersion,
+    latestVersion: getLatestNodeVersion(nodeVersions, nodeMajorVersion),
+    template: getTemplate(variant),
+  };
+};
+
+const isDockerfileOutdated = ({ fileNodeVersion, latestVersion }) => fileNodeVersion
+  !== latestVersion;
+
+const fetchLatestNodeVersions = async () => {
+  const nodeDist = await fetchJson(releaseUrl);
+  return nodeDist.map(({ version }) => version.substring(1));
+};
+
+const findOutdated = async (updateAll) => {
+  const nodeVersions = await fetchLatestNodeVersions();
+
+  const dockerfileMetadatas = getAllDockerfiles(__dirname)
+    .map((file) => getDockerfileMetadata(nodeVersions, file));
+
+  return updateAll
+    ? dockerfileMetadatas
+    : dockerfileMetadatas.filter(isDockerfileOutdated);
+};
+
+const getKeys = (basename) => readFileSync(path.resolve(__dirname, 'keys', basename))
+  .toString().trim().split('\n');
+
+const readTemplate = (template) => readFileSync(
+  path.resolve(__dirname, templateFileMap[template]),
+).toString();
+
+const getBaseImage = ({ template, variant }) => {
+  const tag = template === templates.alpine
+    ? variant.replace(/alpine/, '')
+    : variant;
+
+  return `${templateRepoMap[template]}:${tag}`;
+};
+
+const formatKeys = (keys) => keys.map((key) => `$1${key} \\`).join('\n');
+
+const formatTemplate = (yarnVersion, nodeKeys, yarnKeys, muslChecksum, base, metadata) => {
+  const { latestVersion, template, nodeMajorVersion } = metadata;
+  const baseImage = getBaseImage(metadata);
+  const initialFormat = base.replace(/^FROM.+$/m, `FROM ${baseImage}`)
+    .replace(/^ENV NODE_VERSION .+$/m, `ENV NODE_VERSION ${latestVersion}`)
+    .replace(/^ENV YARN_VERSION .+$/m, `ENV YARN_VERSION ${yarnVersion}`)
+    .replace(/^(\s*)"\${NODE_KEYS\[@]}".*$/m, formatKeys(nodeKeys))
+    .replace(/^(\s*)"\${YARN_KEYS\[@]}".*$/m, formatKeys(yarnKeys));
+
+  if (template !== templates.alpine) {
+    return initialFormat;
+  }
+
+  const pythonVersion = parseInt(nodeMajorVersion, 10) < 14
+    ? 'python2'
+    : 'python3';
+
+  return initialFormat.replace(/\${PYTHON_VERSION}/m, pythonVersion)
+    .replace(/CHECKSUM=CHECKSUM_x64/m, `CHECKSUM="${muslChecksum}"`);
+};
+
+const fetchMuslChecksum = async (nodeVersion) => {
+  const checksums = await fetchText(
+    `https://unofficial-builds.nodejs.org/download/release/v${nodeVersion}/SHASUMS256.txt`,
+  );
+  return checksums.match(/(\S+)\s+\S+-linux-x64-musl.tar.xz/m)[1];
+};
+
+const updateDockerfile = async (yarnVersion, nodeKeys, yarnKeys, metadata) => {
+  const { file, template, latestVersion } = metadata;
+  const base = readTemplate(template);
+  const muslChecksum = await fetchMuslChecksum(latestVersion);
+
+  const formatted = formatTemplate(yarnVersion, nodeKeys, yarnKeys, muslChecksum, base, metadata);
+  writeFileSync(file, formatted);
+};
+
+const updateDockerfiles = async (outdated) => {
+  const yarnVersion = await fetchText(yarnVersionUrl);
```

Comment:
I think here, if it is a security update, it would read the existing Dockerfile, instead of doing the fetch